### PR TITLE
Fixing mysqlconn client library.

### PR DESCRIPTION
**Pull request by alainjobart**
Was not working right for mysql 5.7.
And the test was not passing due to some extra flag.

### CODE REVIEW

File: go/mysqlconn/client_test.go [hunk 1]
```diff
@@ -194,7 +194,11 @@ func TestWithRealDatabase(t *testing.T) {
 		},
 	}
 	if !reflect.DeepEqual(result, expectedResult) {
-		t.Errorf("unexpected result for select, got:\n%v\nexpected:\n%v\n", result, expectedResult)
+		// MySQL 5.7 is adding the NO_DEFAULT_VALUE_FLAG to Flags.
+		expectedResult.Fields[0].Flags |= 4096
```
Comment (michael-berlin) [inline go/mysqlconn/client_test.go]: Can you use a constant here instead?

Comment (michael-berlin) [inline go/mysqlconn/client_test.go]: When you add a flag to the expected value here, how does this test still work for other flavors?

File: go/mysqlconn/client.go [hunk 2]
```diff
@@ -363,13 +363,16 @@ func (c *Conn) parseInitialHandshakePacket(data []byte) (uint32, []byte, error)
 }
 
 func (c *Conn) writeHandshakeResponse41(capabilities uint32, cipher []byte, characterSet uint8, params *sqldb.ConnParams) error {
+	// Build our flags. If the server supported
```
Comment (michael-berlin) [inline go/mysqlconn/client.go]: This new comment should be above the last line, which you added, to be closer to the code?

Comment (alainjobart) [inline go/mysqlconn/client.go]: Done.

File: go/mysqlconn/client_test.go (hunk 1 above)
Comment (alainjobart) [inline go/mysqlconn/client_test.go]: Done, and in a couple other places too.

Comment (alainjobart) [inline go/mysqlconn/client_test.go]: expectedResult is only use for this comparison here, not later on by other flavors.

**Comment by michael-berlin** [comment]
LGTM

**Pull request merged**